src/lib/frontend/d_state_option.ml

```diff
@@ -0,0 +1,91 @@
+(**************************************************************************)
+(*                                                                        *)
+(*     Alt-Ergo: The SMT Solver For Software Verification                 *)
+(*     Copyright (C) 2013-2023 --- OCamlPro SAS                           *)
+(*                                                                        *)
+(*     This file is distributed under the terms of OCamlPro               *)
+(*     Non-Commercial Purpose License, version 1.                         *)
+(*                                                                        *)
+(*     As an exception, Alt-Ergo Club members at the Gold level can       *)
+(*     use this file under the terms of the Apache Software License       *)
+(*     version 2.0.                                                       *)
+(*                                                                        *)
+(*     ---------------------------------------------------------------    *)
+(*                                                                        *)
+(*     The Alt-Ergo theorem prover                                        *)
+(*                                                                        *)
+(*     Sylvain Conchon, Evelyne Contejean, Francois Bobot                 *)
+(*     Mohamed Iguernelala, Stephane Lescuyer, Alain Mebsout              *)
+(*                                                                        *)
+(*     CNRS - INRIA - Universite Paris Sud                                *)
+(*                                                                        *)
+(*     Until 2013, some parts of this code were released under            *)
+(*     the Apache Software License version 2.0.                           *)
+(*                                                                        *)
+(*     ---------------------------------------------------------------    *)
+(*                                                                        *)
+(*     More details can be found in the directory licenses/               *)
+(*                                                                        *)
+(**************************************************************************)
+
+module O = Options
+module State = D_loop.State
+module Typer = D_loop.Typer
+
+module type S = sig
+  type k
+
+  type t
+
+  val set : k -> Typer.state -> Typer.state
+
+  val get : Typer.state -> t
+
+  val reset : Typer.state -> Typer.state
+end
+
+let create_opt
+    (type k)
+    (type t)
+    ?(on_update=ignore)
```

Comment:
This seems to still be unused.

Reply:
Disregard - did not see your reply on the other comment

Reply:
> It is used in set and, while there is no option setting on_update yet, it will be useful for other options (like the steps limit that has to check a few things before setting the option).